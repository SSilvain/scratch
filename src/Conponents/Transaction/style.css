.transaction{
    background: #f2f2ff;
    border-color: #bdbdbd;
    border-radius: 15px;
    padding: 5px;
}