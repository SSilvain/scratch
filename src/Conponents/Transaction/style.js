import styled from "styled-components";

export const Wrapper = styled.div`
    background: ${({ value }) => (value > 0 ? "#ccc" : "#eee")};
    border-color: #bdbdbd;
    border-radius: 5px;
    padding: 15px 25px;
`;

Wrapper.displayName = "TransactionWrapper";
