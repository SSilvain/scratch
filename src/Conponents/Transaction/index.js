import { Wrapper, TransactionDate, Value, Comment } from "./style"

const Transaction = ({ value, date, comment }) => {
    value = +value;
    return (
        <Wrapper value={value}>
            <TransactionDate>{date}</TransactionDate>
            <Value>{value.toFixed(2)}</Value>
            <Comment>{comment}</Comment>
        </Wrapper>
    );
};

export default Transaction;
