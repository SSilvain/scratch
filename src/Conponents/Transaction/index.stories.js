// Transaction.stories.js

import React from 'react';

import Transaction from '.';

//👇 This default export determines where your story goes in the story list
export default {
  title: 'Transaction',
  component: Transaction,
};

//👇 We create a “template” of how args map to rendering
const Template = (args) => <Transaction {...args} />;

export const FirstStory = Template.bind({});

FirstStory.args = {
    label: "some action",
    value: 0
  /*👇 The args you need here will depend on your component */
};