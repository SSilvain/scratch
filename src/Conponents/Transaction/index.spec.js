import React from "react";
import {shallow} from "enzyme";
import Transaction from ".";

describe("Transaction component", () =>{
    it("should show transaction", ()=>{
        const props = {
                value: "22",
                comment: "hell commnt",
                date: "10.10.21"
        }
        const sut = shallow(<Transaction {...props}/>);
        expect(sut).toMatchSnapshot();
    })
})