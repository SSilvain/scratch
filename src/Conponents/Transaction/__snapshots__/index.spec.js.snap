// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should show transaction 1`] = `
<Comment
  value={22}
>
  <styled.div>
    10.10.21
  </styled.div>
  <styled.div>
    22.00
  </styled.div>
  <styled.div>
    hell commnt
  </styled.div>
</Comment>
`;
